python/pip_install/extract_wheels/lib/arguments_test.py

```diff
@@ -0,0 +1,24 @@
+import unittest
+import argparse
+
+from python.pip_install.extract_wheels.lib import arguments
+
+
+class ArguementsTestCase(unittest.TestCase):
+    def test_arguments(self) -> None:
+        parser = argparse.ArgumentParser()
+        parser = arguments.parse_common_args(parser)
+        repo_name = "foo"
+        index_url = "--index_url=pypi.org/simple"
+        args_dict = vars(parser.parse_args(
+            args=["--repo", repo_name, "--extra_pip_args={index_url}".format(index_url=index_url)]))
```

Comment:
Hey, thanks for the feature, we are already trying to use it from the PR directly and it's a game-changer!

**Problem**: we use our own PyPi index in requirements.txt (--index-url=... at the top of the requirements file) and it seems like it is being ignored (I can be wrong or misusing the tool), but if that is the case, it would be great if 
that is passed to extra_pip_args automatically.

Also when we try to pass the pypi url via `pip_extra_args` explicitly, like this:
```
pip_parse(
    name = "my_repo",
    quiet = False,
    timeout = 1500,
    requirements_lock = "@//.../requirements-lock.txt",
    python_interpreter = "python3.7",
    extra_pip_args=["--index-url=http://<pypi-host>/repository/pypi-all/simple", "--trusted-host=<pypi-host>"]
)
```

We get an error, like this:
```
ERROR: .../WORKSPACE:50:13: //external:repo_pypi__setuptools: expected value of type 'list(string)' for attribute 'extra_pip_args' in 'whl_library' rule, but got "{\"args\":[\"--index-url=http://.../repository/pypi-all/simple\"]}" (string)
```

Reply:
Hey thanks for the feedback, I probably missed the --index-url argument. I'll try to repro the issue with extra_pip_args as well.

Reply:
I fixed the issue with --extra_pip_args not being passed to the child repos correctly. I will leave the issue of pip flags in a requirements.txt file as a TODO, as this is an experimental rule for the moment.